### ISSUE TITLE
Status and.. stuff

### ISSUE THREAD

**Issue by patternseek** [JanMiksovsky/polymer-micro-test#1]
Hi,

I'm really quite excited about this experiment as it pretty much includes everything I want from Polymer without including all the heavier duty stuff and becoming too dependent on their various abstractions.

I've made a couple of changes to a local copy, one allowing the creation of a component with no shadow root and the other changing the call to this._marshalHostAttributes() to this._takeAttributesToModel(this); as the former causes breakage if the component is used in a page where other components are using full Polymer.

The only outstanding issue I have is that relative URIs in src attributes (for images in my case) aren't being modified by the HTML Imports polyfill for some reason.

Could you give me an idea where you are with this idea at the moment? Your blog post mentioned you're in discussions with others, are you getting any traction?

By the way I'd like to actually use this in a live project but there's no license file or license declarations in the files. Could you add one by any chance?

**Comment by JanMiksovsky** [comment]
Hi @patternseek! Thanks for your interest and questions.

We've continued to blog about the evolution of our thinking here. Our current thinking is captured in our latest post on [building web components from a loose framework of mixins](https://component.kitchen/blog/posts/building-web-components-from-a-loose-framework-of-mixins). That is directly inspired by this polymer-micro-test experiment, but takes it further by stepping away from Polymer entirely. We like most of what polymer-micro provides, but even polymer-micro feels a bit too heavy for us. In order to reduce abstractions, we're trying to see whether we can break the core polymer-micro feature set into separable pieces. Those are the mixins we're pursuing in the [core-component-mixins](https://github.com/ComponentKitchen/core-component-mixins) repo referenced by the blog post.

This pure mixin approach feels promising enough that we're focusing our efforts on that for the time being. We think a lot of good work has gone into polymer-micro, so we're trying to learn from them where we can. I've mostly been using this polymer-micro-test repository as a sort of lab experiment where I can watch Polymer do its work in a more controlled environment without the distractions of full Polymer. But at the moment at least, I'm not actively developing this.

I hope our evolving work at the component framework level isn't too confusing. We're trying to figure things out as we go along, and our plans change as we learn. If you're interested in creating web components without Polymer, I'd love to hear what you think about our latest direction. Keep in mind that it's not yet a stable model for component development. We're trying very hard to collaborate with others on it, and it's likely to continue to change for a while.

I'm happy to do what I can to make this polymer-micro-test experiment more useful to you and others. Thanks for raising the point about _takeAttributesToMode — I hadn't gotten around to trying a mix of our polymer-micro components with full Polymer. If you'd like to make a PR against this repo to fix that, I'll happily take that.

I hadn't noticed that relative URI issue either. You might try filing an issue on the Polymer repo. The Polymer team is actively reconsidering the way they factor their library into layers, and I expect it would be useful for them to know where polymer-micro falls short. (Our latest work isn't using HTML Imports, so it won't have that bug — but neither will it provide any help resolving relative URIs. We're still trying to work out a good solution for that.)

By all means feel free to use the code here in polymer-micro-test in your own project. To document and clarify your rights to do so, I've added a MIT license to the project. I was kind of surprised there wasn't one there already — thanks for pointing out that omission.

**Comment by patternseek** [comment]
I've looked at your mixin plans and they make a lot of sense. I had noticed the post but didn't quite clock that it was a progression from the polymer-micro idea.

I'll add a PR for the _takeAttributesToMode() change when I get a minute, I'm very behind on my current project just now. Also I added optional support for creating a component with no shadow root as having an enforced shadow is a bit limiting at the moment while so much existing JS is written without them in mind. Shall I put in a PR for that too? 

I realise this probably isn't a project you're going to continue with but it's enough to give me something that's ready to work with now that allows me to build on Polymer's base without going all in on the larger library which is a very useful stopgap as I have a deadline to get something out the door.

Thanks for the license update.
